pennylane/interfaces/execution.py

```diff
@@ -360,6 +360,22 @@ def cost_fn(params, x):
         for i, tape in enumerate(tapes):
             tapes[i] = expand_fn(tape)
 
+        if gradient_kwargs.get("method", "") == "adjoint_jacobian":
+            # adjoint jacobian needs further expansion
+            # this is a quick patch that should be cleaned up by later refactoring
+            if INTERFACE_MAP[interface] == "jax":
+                stop_at = (
+                    ~qml.operation.is_measurement & qml.operation.has_nopar
+                    | qml.operation.has_unitary_gen

```

Comment:
Switching to `qml.operation.has_unitary_gen` instead of `qml.operation.has_gen` gave a 30% decrease in the time it took to calculate the gradient.

This suggests that the `try-expect` logic way of determining properties is *extremely* inefficient.